ui/components/multichain/account-details/account-details.tsx

```diff
@@ -51,8 +62,18 @@ export const AccountDetails = ({ address }) => {
   const {
     metadata: { name },
   } = useSelector((state) => getInternalAccountByAddress(state, address));
+  const keyrings: {
+    accounts: string[];
+    metadata: KeyringMetadata;
+  }[] = useSelector(getMetaMaskKeyrings);
+  const keyringId = keyrings.find((kr) => kr.accounts.includes(address))
+    ?.metadata.id;
```

Comment:
Similar to the previous PR, this is not always guaranteed to have a HD keyring here.

We should move this function: https://github.com/MetaMask/metamask-extension/blob/ebfd25e7fc5923f4b39dab206add5eca4675ce09/app/scripts/metamask-controller.js#L4287-L4304

Into the `shared/` folder and use it there too IMO.